### PR TITLE
Make ClusterCaRenewalTest test with real existing certificates

### PR DESCRIPTION
**Pull request by tinaselenge**
### Type of change

- Refactoring

### Description

This PR refactors ClusterCaRenewalTest to test with real existing certificates so that checks for expiration and subject change are tested instead of mocking them to trigger certifcate generation. 

getSubjectAltNames method of ClusterCa is updated with null check for alternate subject names as that could cause NPE which then gets ignored.  It is also updated to return empty list instead of null, as it causes tests to fail because of regenerating certificates unnecessarily due to null != [] being treated as "subjects changed" in certSubjectChanged method:
```
2026-07-01 16:59:19 INFO  Ca:400 - Reconciliation #0(test) kind(namespace/name): Current alternate subjects: null
2026-07-01 16:59:19 INFO  Ca:401 - Reconciliation #0(test) kind(namespace/name): Desired alternate subjects: []
``` 

changedSubjectOfCertificates test is updated so that DNS name is changed instead of the common name, as that's the expected behaviour in the ClusterCa. 

Mock override for addKeyAndCertToKeyStore() is removed, as this is only used by ClientsCa (KafkaUserModel) so it's not necessary for this test class. 

Mocked generateSignedCert method is left to create not real certificates, as we just care that this method is called and new certificates don't need to be real. 
  
### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update documentation
- [ ] Update CHANGELOG.md (if present)
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Try your changes inside a Kubernetes cluster, not just from unit tests
- [ ] AI assistance was used to create this PR (see the [Strimzi AI policy](https://github.com/strimzi/governance/blob/main/AI_POLICY.md))

### CODE REVIEW

**Review by katheris — APPROVED** [review]
Thanks @tinaselenge LGTM

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java [hunk 1]
```diff
@@ -411,17 +411,18 @@ private boolean hasCaCertGenerationChanged(int certAndKeyCaCertGeneration, Strin
...
             Collection<List<?>> altNames = cert.getSubjectAlternativeNames();
-            subjectAltNames = altNames.stream()
-                    .filter(name -> name.get(1) instanceof String)
-                    .map(item -> (String) item.get(1))
-                    .collect(Collectors.toList());
+            if (altNames != null) {
+                subjectAltNames = altNames.stream()
+                        .filter(name -> name.get(1) instanceof String)
+                        .map(item -> (String) item.get(1))
+                        .collect(Collectors.toList());
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java:423]: Any chance we could include this in the test? (I guess by using some certificate with SANs as well?)

Comment (tinaselenge) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java:423]: yes, we can do that. Maybe it's actually more realistic to test server certificates with SAN.

**Review by scholzj — APPROVED** [review]
Thanks. LGTM.
